Build failure: mozillavpn

### Steps To Reproduce

Steps to reproduce the behavior:
1. `inputs.nixpkgs.url = flake:nixpkgs/nixos-unstable;`
1. `services.mozillavpn.enable = true;`
1. `sudo nixos-rebuild dry-activate`

### Build log

```
In file included from /build/source/src/platforms/linux/daemon/dnsutilslinux.h:12,
                 from /build/source/src/platforms/linux/daemon/dbusservice.h:13,
                 from /build/source/src/platforms/linux/daemon/dbusservice.cpp:5:
/build/source/src/platforms/linux/daemon/dbustypeslinux.h: In function 'const QDBusArgument& operator>>(const QDBusArgument&, DnsResolver&)':
/build/source/src/platforms/linux/daemon/dbustypeslinux.h:49:26: error: 'qsizetype QByteArray::count() const' is deprecated: Use size() or length() instead. [-Werror=deprecated-declarations]
   49 |     } else if (data.count() >= 4) {
      |                ~~~~~~~~~~^~
In file included from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qmetatype.h:11,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtDBus/qtdbusglobal.h:8,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtDBus/qdbuscontext.h:7,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtDBus/QDBusContext:1,
                 from /build/source/src/platforms/linux/daemon/dbusservice.h:8:
/nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qbytearray.h:501:22: note: declared here
  501 |     inline qsizetype count() const noexcept { return size(); }
      |                      ^~~~~
```

Full logs: https://gist.github.com/devurandom/9b6d29eb81dabb7de6ad8f4a02a2c69b

### Additional context

My `flake.lock`:
```json
{
  "nodes": {
    "home-manager": {
      "inputs": {
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1729260213,
        "narHash": "sha256-jAvHoU/1y/yCuXzr2fNF+q6uKmr8Jj2xgAisK4QB9to=",
        "owner": "nix-community",
        "repo": "home-manager",
        "rev": "09a0c0c02953318bf94425738c7061ffdc4cba75",
        "type": "github"
      },
      "original": {
        "id": "home-manager",
        "type": "indirect"
      }
    },
    "nixpkgs": {
      "locked": {
        "lastModified": 1729070438,
        "narHash": "sha256-KOTTUfPkugH52avUvXGxvWy8ibKKj4genodIYUED+Kc=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "5785b6bb5eaae44e627d541023034e1601455827",
        "type": "github"
      },
      "original": {
        "id": "nixpkgs",
        "ref": "nixos-unstable",
        "type": "indirect"
      }
    },
    "root": {
      "inputs": {
        "home-manager": "home-manager",
        "nixpkgs": "nixpkgs"
      }
    }
  },
  "root": "root",
  "version": 7
}
```

### Notify maintainers

@andersk

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.1, NixOS, 24.11 (Vicuna), 24.11.20241004.bc947f5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/sdzpqjwx7pdx6lsq6llyfqqf7hspp83c-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/mo/mozillavpn/package.nix
+++ b/pkgs/by-name/mo/mozillavpn/package.nix
@@ -4,6 +4,7 @@
   cargo,
   cmake,
   fetchFromGitHub,
+  fetchpatch,
   go,
   lib,
   libcap,
@@ -30,7 +31,23 @@ stdenv.mkDerivation (finalAttrs: {
     fetchSubmodules = true;
     hash = "sha256-X2rtHAZ9vbWjuOmD3B/uPasUQ1Q+b4SkNqk4MqGMaYo=";
   };
-  patches = [ ];
+  patches = [
+    # Fix build errors from deprecated QByteArray::count()
+    (fetchpatch {
+      url = "https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9961/commits/1b358d27d4bf29567b5d58f3591146bf639b99e1.patch";
+      hash = "sha256-LeDgwZaQDgS8HNf9k2fC0RYQy4nGEq0DMNjY7muNads=";
+    })
+    # Fix build errors from deprecated QVariant::type()
+    (fetchpatch {
+      url = "https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9961/commits/ebdd38ce19ef6eb80f076acf93299bd7d24ae6db.patch";
+      hash = "sha256-ZWl0wHH5Foxlttj/GK5phr/C6qJv39U2GWIofZR+Rto=";
+    })
+    # Fix build errors from deprecated QEventPoint::pos and friends
+    (fetchpatch {
+      url = "https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9961/commits/10b1c98517dac4eacffd6890c551b817aedd4a19.patch";
+      hash = "sha256-DHOtvVDEdQ+k2ggg4HGpcv1EmKzlijNRTi1yJ7a1bWU=";
+    })
+  ];
 
   netfilter = buildGoModule {
     pname = "${finalAttrs.pname}-netfilter";

